test/integration/testStreamRows.js

```diff
@@ -462,3 +464,93 @@ describe('Test Stream Rows HighWaterMark', function ()
     });
   });
 });
+
+describe('Test stream row modes', function () {
+  this.timeout(18000);
+  let connection;
+  const sql = 'select * ' +
+    'from (select \'a\' as key, 1 as foo, 3 as name) as table1 ' +
+    'join (select \'a\' as key, 2 as foo, 3 as name2) as table2 on table1.key = table2.key ' +
+    'join (select \'a\' as key, 3 as foo) as table3 on table1.key = table3.key'
+
+
+  const expectedArray = ['a', 1, 3, 'a', 2, 3, 'a', 3]
+  const expectedObject = {KEY_0: 'a', FOO_1: 1, NAME: 3, KEY_3: 'a', FOO_4: 2, NAME2: 3, KEY_6: 'a', FOO_7: 3}
+
+  const createItCallback = function (testCase) {
+    return function (done) {
+      this.timeout(180000);
+      connection.execute({
+        sqlText: sql,
+        streamResult: true,
+        complete: function (err, stmt) {
+          testUtil.checkError(err);
+          var stream = stmt.streamRows({rowMode: testCase.rowMode});
+          stream.on('data', function (row) {
+            const mode = testCase.rowMode || testCase.connectionRowMode
+            if (mode === 'array') {
```

Comment:
are we able to get rid of this if and always test with deepStrictEqual?

Reply:
we can. Fixed